src/i_winmusic.c

```diff
@@ -1552,7 +1566,11 @@ static void I_WIN_PauseSong(void *handle)
         return;
     }
 
+    EnterCriticalSection(&CriticalSection);
+
     win_midi_state = STATE_PAUSING;
+
+    LeaveCriticalSection(&CriticalSection);
 }
 
 static void I_WIN_ResumeSong(void *handle)
```

Comment:
Should `I_WIN_PauseSong()` be replaced with something close to `I_WIN_StopSong()` now? Pausing would be instant instead of delayed until the next callback.

Similarly, should `I_WIN_ResumeSong()` be replaced with something close to `I_WIN_PlaySong()` ?

Reply:
Honestly, pause is a very obscure feature. I like the way it works right now. If you want to change it, feel free to do so.

Reply:
You're right. I'll look at it in a separate PR.